Libraries/LibWeb/DOM/CanvasRenderingContext2D.cpp

```diff
@@ -77,6 +98,14 @@ void CanvasRenderingContext2D::translate(double x, double y)
     m_translate_y = y;
 }
 
+Gfx::Rect CanvasRenderingContext2D::compute_rect(int x, int y, int width, int height)
+{
+    return Gfx::Rect((x + m_translate_x) * m_scale_x,
+                     (y + m_translate_y) * m_scale_y,
+                     width * m_scale_x,
+                     height * m_scale_y);
```

Comment:
Transforms are supposed to accumulate, so I we're gonna need to keep an affine transformation matrix and update it on calls to `scale()` and `translate()`. It will also allow us to implement `transform()` correctly. :)

Reply:
Actually let's just merge this as-is and we can add transforms separately.